File: 2558-take-gifts-from-the-richest-pile/2558-take-gifts-from-the-richest-pile.cpp
class Solution {
public:
    long long pickGifts(vector<int>& arr, int k) {
        priority_queue<long long> pq;
        long long ans=0;
        for(auto a: arr)pq.push(a);
        
        while(k--){
            pq.push(sqrt(pq.top()));
            pq.pop();  
        }
        while(!pq.empty()){
            ans+=pq.top();
            pq.pop();
        }
        return ans;
    }
};

File: 2558-take-gifts-from-the-richest-pile/2558-take-gifts-from-the-richest-pile.py
class Solution:
	def pickGifts(self, gifts: List[int], k: int) -> int:
		while k > 0:
			max_pile = max(gifts)
			index = gifts.index(max_pile)
			gifts[index] = int(math.sqrt(max_pile))
			k = k - 1
		return sum(gifts)